src/applications/facility-locator/tests/e2e/search.cypress.spec.js

```diff
@@ -1,33 +1,70 @@
-import path from 'path';
+import { facilityTypesOptions } from '../../config';
+import { LocationType } from '../../constants';
 
-describe('Facility Search', () => {
-  before(() => {
-    cy.syncFixtures({
-      constants: path.join(__dirname, '..', '..', 'constants'),
-    });
-  });
+describe('Facility search', () => {
+  it('does a simple search and finds a result on the list', () => {
+    cy.visit('https://staging.va.gov/find-locations/'); // can we make it real e2e?, with live data
```

Comment:
This makes this test into a regression test. Maybe that's OK, but it will not catch any bugs we introduce into our feature branches.

Reply:
@U-DON we are authenticated app, can we just use staging for our target environment ? or the review instance http://20733db61b9b3535298b7a1504932cf4.review.vetsgov-internal/ for example ?

Reply:
We should stick to running against `localhost`. These are not intended to be integration tests. They also run on every build, and we don't want to send all these requests to staging as a side effect of our CI and potentially DDOS staging.

If you'd like to write a test that hits staging that you can run yourself outside of CI, I would recommend renaming the test without the `.cypress.spec.js` so it doesn't get picked up by the pattern matching.

Reply:
I don't know about DDOS staging, but if these are not real end to end I would say these tests are not very useful for us. The fact that we have maintain the mock data is also unnecessary in our case because our data lives in several backend services through lighthouse or ppms. Testing against live data is how we can verify for potential data structure changes.

Reply:
What about testing against the review instance?

Reply:
I would say there's still value in testing that the front-end app functions as intended, isn't there?

It seems like the unreliable data structure problem is symptomatic of a larger problem and can't be simply resolved by pointing the browser tests to staging. Staging also isn't intended to be an automated test environment.

If the data structure changes and fail the test, it may be a useful signal to your team in some ways, but we shouldn't have the test break the build for every other team either. That's one of the biggest Issues I see with tying the success of our builds/tests to third party services. We should be focused on testing our own code.

Reply:
I see your point about having to maintain mock data. We're working on leveraging Pact into our testing processes to maintain a stabler integration between FE and BE.

Unfortunately, I don't see us supporting testing against a deployed environment within our CI. There are lots of ramifications to that, and we'd need to discuss with multiple stakeholders on the platform to determine its feasibility.

Reply:
As for review instances, that's not a reliable environment to test against in CI either. They're temporarily deployed environments, and once they're discarded, the tests will start breaking.

As I mentioned, if you insist on having tests that run against a deployed environment, maybe you can have a separate test that you can run locally on demand as long as it's not named in a way that includes it in CI. Maybe a suffix like `.cypress.regression.js`.

Reply:
@U-DON . Alright, I put the mock data back and hitting localhost, however it has failed, can you help me understand what I am doing wrong here ? https://app.circleci.com/pipelines/github/department-of-veterans-affairs/vets-website/10642/workflows/70a18749-a726-49ba-8889-a1f0a7061fed/jobs/75172/tests

Reply:
It seems you've changed the stub to be for `'/v1/facilities/va?*'`, but it's still making a request to `v0`. Note that CI is running against a production build. It could be that you're setting an environment flag somewhere to differentiate between the API versions.

Reply:
You can see which responses haven't been stubbed by looking for XHR vs XHR stubbed in the list of steps on the left sidebar.

Reply:
I tried changing it to `/v0/facilities/va?*`, but it looks like you'll still need to create a stub for `/v0/facilities/va/vha_674BY` to proceed in the breadcrumbs test.

Speaking of breadcrumbs, it may pass in Jenkins because it's able to run the full content build and generate the real breadcrumbs, but you may find it failing in Circle since that doesn't have access to the content build and has to imitate the landing page. **If it the Circle tests end up failing to match the breadcrumbs**, I would suggest changing the test to more generally test that the details page of a facility looks correct instead of verifying the exact breadcrumbs.

Reply:
<img width="659" alt="Screen Shot 2020-08-06 at 11 41 08 AM" src="https://user-images.githubusercontent.com/100468/89558518-ec445e00-d7d9-11ea-8d61-da9e5a3e14d3.png">
Tests pass locally though, why?

Reply:
is it because the tests use the MockApi?

Reply:
Most likely, yes.

Reply:
Ok makes sense. @U-DON, how do create the second stub  for `/v0/facilities/va/vha_674BY`?

Currently I have:   but still fails to fetch the facility 

 ```beforeEach(() => {
    cy.server();
    cy.route('GET', '/v0/feature_toggles?*', []);
    cy.route('GET', '/v0/maintenance_windows', []);
    cy.route('GET', '/v0/facilities/va?*', 'fx:constants/mock-facility-data');
    cy.route(
      'GET',
      '/v0/facilities/va?*',
      'fx:constants/mock-facility-data-v1',
    );
    cy.route(
      'GET',
      '/v0/facilities/va/vha_674BY',
      'fx:constants/mock-facility-v1',
    );
    cy.route('GET', '/geocoding/**/*', 'fx:constants/mock-geocoding-data');
  });

Reply:
The mock response you're passing looks like an array of facilities. The `FacilityDetail` component  is expecting a single facility. I think what's happening is that the `MockApi` was just filtering one facility out of the list response to build the mock response. So `mock-facility-data.json` (and the v1 version)  isn't actually mock data for one facility, but for a list of facilities.

Reply:
Sorry, I was mistaken. It looks like you need to stub out the `feature_toggles` response to include the name of your feature. Right now it's returning an empty array.

Reply:
but I am overriding the mock data before rendering `FacilityDetail`
```cy.route(
      'GET',
      '/v0/facilities/va/vha_674BY',
      'fx:constants/mock-facility-v1',
    ).as('fetchFacility');
    //    cy.get('.facility-result a') // I thought this would visit the page?
    //       .first()
    //       .click();
    // But visit instead?
    cy.visit('/find-locations/facility/vha_674BY');

    cy.get('.all-details').should('exist');```

Reply:
As for the feature toggles - I am waiting for @mpelzsherman to merge https://github.com/department-of-veterans-affairs/vets-website/pull/13714 so that I can use directly v1 without a feature flag.

Reply:
I think I found the issue and got it to pass with my local production build. You needed to wrap the entire facility response under a `data` attribute.

Reply:
All good now. E2E tests are all passing. But now it looks like one of your unit tests is failing.

Reply:
Yeah, I can fix that. Thanks!